src/main/javac/org/apache/brooklyn/tosca/a4c/platform/transformer/ToscaPlanToSpecTransformer.java

```diff
@@ -243,13 +276,18 @@ private static boolean isTosca(Map<?,?> obj) {
                 }
 
                 if (hostNodeId!=null) {
-                    EntitySpec<?> parent = topLevelNodeSpecs.get(hostNodeId);
-                    if (parent!=null) {
-                        parent.child(thisNode);
-                        parent.configure(SoftwareProcess.CHILDREN_STARTABLE_MODE, ChildStartableMode.BACKGROUND_LATE);
-                    } else {
-                        throw new IllegalStateException("Can't find parent '"+hostNodeId+"'");
+                    //EntitySpec<?> parent = topLevelNodeSpecs.get(hostNodeId);
+                    //if (parent!=null) {
+                    //    parent.child(thisNode);
+                    //    parent.configure(SoftwareProcess.CHILDREN_STARTABLE_MODE, ChildStartableMode.BACKGROUND_LATE);
+                    //} else {
+                    //    throw new IllegalStateException("Can't find parent '"+hostNodeId+"'");
+                    //}
```

Comment:
If the targer NodeTemplate is not process still, a NullPointerException will be thrown. Then, the children are added to a hosted list, and when all templates will be processed, the child will be added to their parent.